packages/vite/src/node/utils.ts

```diff
@@ -139,20 +140,51 @@ export function createDebugger(
   }
 }
 
+function testCaseInsensitiveFS() {
+  if (!CLIENT_ENTRY.endsWith('client.mjs')) {
+    throw new Error(
+      `cannot test case insensitive FS, CLIENT_ENTRY const doesn't contain client.mjs`
+    )
+  }
+  if (!fs.existsSync(CLIENT_ENTRY)) {
+    throw new Error(
+      'cannot test case insensitive FS, CLIENT_ENTRY does not point to an existing file: ' +
+        CLIENT_ENTRY
+    )
+  }
+  return fs.existsSync(CLIENT_ENTRY.replace('client.mjs', 'cLiEnT.mjs'))
+}
+
+export const isCaseInsensitiveFS = testCaseInsensitiveFS()
+
 export const isWindows = os.platform() === 'win32'
+
 const VOLUME_RE = /^[A-Z]:/i
 
 export function normalizePath(id: string): string {
-  return path.posix.normalize(isWindows ? slash(id) : id)
+  let normalized = path.normalize(id)
+  if (isCaseInsensitiveFS) {
+    normalized = normalized.toLowerCase()
+  }
```

Comment:
I think we shouldn't do the normalization to lowercase in this PR. We had a few requests to do so before but decided that it was better to ask the user to properly name paths. Something may work in windows but fail in linux if we had this in. Touching `normalizePath` is also quite a big change. Let's check the rest of the issues without it